proto/concept.proto

```diff
@@ -0,0 +1,963 @@
+        }
+        message Res {}
+    }
+
+    message GetSupertypes {
+        message Req {}
+        message ResPart {
+            repeated RelationType relation_types = 1;
+        }
+    }
+
+    message GetSubtypes {
+        message Req {
+            Type.Transitivity transitivity = 1;
+        }
+        message ResPart {
+            repeated RelationType relation_types = 1;
+        }
+    }
+
+    message GetInstances {
+        message Req {
+            Type.Transitivity transitivity = 1;
+        }
+        message ResPart {
+            repeated Relation relations = 1;
+        }
+    }
+
+    message GetRelates {
+        message Req {
+            Type.Transitivity transitivity = 1;
+        }
+        message ResPart {
+            repeated RoleType role_types = 1;
+        }
+    }
+
+    message GetRelatesForRoleLabel {
+        message Req {
+            string label = 1;
+        }
+        message Res {
+            optional RoleType role_type = 1;
+        }
+    }
+
+    message GetRelatesOverridden {
+        message Req {
+            string label = 1;
+        }
+        message Res {
+            optional RoleType role_type = 1;
+        }
+    }
+
+    message SetRelates {
+        message Req {
+            string label = 1;
+            optional string overridden_label = 2;
+        }
+        message Res {}
+    }
+
+    message UnsetRelates {
+        message Req {
+            string label = 1;
+        }
+        message Res {}
+    }
+}
+
+message AttributeType {
+    string label = 1;
+    ValueType value_type = 2;
+    bool is_root = 3;
+    bool is_abstract = 4;
+
+    enum ValueType {
+        OBJECT = 0;
+        BOOLEAN = 1;
+        LONG = 2;
+        DOUBLE = 3;
+        STRING = 4;
+        DATETIME = 5;
+    }
+
+    message Put {
+        message Req {
+            Attribute.Value value = 1;
+        }
+        message Res {
+            Attribute attribute = 1;
+        }
+    }
+
+    message Get {
+        message Req {
+            Attribute.Value value = 1;
+        }
+        message Res {
+            optional Attribute attribute = 1;
+        }
+    }
+
+    message GetSupertype {
+        message Req {}
+        message Res {
+            AttributeType attribute_type = 1;
+        }
+    }
+
+    message SetSupertype {
+        message Req {
+            AttributeType attribute_type = 1;
+        }
+        message Res {}
+    }
+
+    message GetSupertypes {
+        message Req {}
+        message ResPart {
+            repeated AttributeType attribute_types = 1;
+        }
+    }
+
+    message GetSubtypes {
+        message Req {
+            Type.Transitivity transitivity = 1;
+            optional ValueType value_type = 2;
+        }
+        message ResPart {
+            repeated AttributeType attribute_types = 1;
+        }
+    }
+
+    message GetInstances {
+        message Req {
+            Type.Transitivity transitivity = 1;
+            optional ValueType value_type = 2;
```

Comment:
here too, and in `GetOwners`. Anywhere it'ss a restriction? I wonder if that's all the time so it becomes redundant.

Reply:
It's always a restriction, outside of `PutAttributeType`. I think tacking on `_filter` would be redundant.